game-core/src/test/java/games/strategy/triplea/attachments/TriggerAttachmentTest.java

```diff
@@ -87,4 +94,39 @@ void testTriggerProductionFrontierEditChange() {
     assertTrue(allValues.stream().anyMatch(s -> s.contains("rule3") && s.contains("added")));
     assertTrue(allValues.stream().allMatch(s -> s.contains("frontier")));
   }
+
+  @Test
+  void testTriggerPlayerPropertyChange() throws MutableProperty.InvalidValueException {
+
+    final GameData gameData = bridge.getData();
+    final TriggerAttachment triggerAttachment = mock(TriggerAttachment.class);
+    final Set<TriggerAttachment> satisfiedTriggers = ImmutableSet.of(triggerAttachment);
+
+    when(triggerAttachment.getPropertyMap()).thenCallRealMethod();
+    triggerAttachment.getPropertyMap().get("playerAttachmentName").setValue("rulesAttachment:RulesAttachment");
+    when(triggerAttachment.getPlayerProperty())
+        .thenReturn(Arrays.asList(
+            Tuple.of("productionPerXTerritories", "someNewValue")));
+    when(triggerAttachment.getName()).thenReturn("mockedTriggerAttachment");
+
+    final PlayerId playerId = mock(PlayerId.class);
+    when(playerId.getAttachment("rulesAttachment"))
+        .thenReturn(new RulesAttachment(null, null, gameData));
+    when(triggerAttachment.getPlayers()).thenReturn(Arrays.asList(playerId));
+
+
+    TriggerAttachment.triggerPlayerPropertyChange(
+        satisfiedTriggers,
+        bridge,
+        "beforeOrAfter",
+        "stepName",
+        false, // useUses
+        false, // testUses
+        false, // testChance
+        false); // testWhen
+    final ArgumentCaptor<CompositeChange> argument = ArgumentCaptor.forClass(CompositeChange.class);
+    verify(bridge).addChange(argument.capture());
+    final CompositeChange change = argument.getValue();
+    assertFalse(change.isEmpty());
```

Comment:
Using ArgumentCaptor is probably overkill at this point.
A simple
```java
verify(bridge).addChange(ArgumentMatchers.eq(Collections.emptySet()));
```
Might already do the trick (I'm only 80% sure tbh, but it's less code in case that works).
Note that `ArgumentMatchers.eq` can be statically imported.

Reply:
You are right, it was overkill, after I looked into it this sufficed: `verify(bridge).addChange(argThat((arg) -> !arg.isEmpty()));`

Reply:
Thanks for checking. Looks like I misinterpreted the code at this point, so finding another way to do this this way is awesome 👍